### PR TITLE
core: add more PDL dialect constructors

### PR DESCRIPTION
**Pull request by superlopuh**

### CODE REVIEW

**Review by martin-luecke — APPROVED** [review]
LGTM! 
especially not having `sym_name: OptOpAttr[StringAttr]` bothered me a lot

File: xdsl/dialects/builtin.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def from_enum(signedness: Signedness) -> SignednessAttr:
 
 
 @irdl_attr_definition
-class IntegerType(ParametrizedAttribute):
+class IntegerType(ParametrizedAttribute, MLIRType):
     name: str = "integer_type"
     width: ParameterDef[IntAttr]
     signedness: ParameterDef[SignednessAttr]
```
[the remark this answers is not in the document]
Comment (math-fehr) [reply xdsl/dialects/builtin.py:244]: The problem is that we want compatibility with MLIR, and this means that types should be prefixed with `!` when parsed/printed in MLIR.
We do not want to make a difference between types and attributes, but we still need to only for that case.

File: tests/dialects/test_pdl.py [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+from xdsl.dialects.builtin import ArrayAttr, IntegerAttr, StringAttr
+from xdsl.utils.exceptions import VerifyException
 
 type_type = pdl.TypeType()
 attribute_type = pdl.AttributeType()
+value_type = pdl.ValueType()
+operation_type = pdl.OperationType()
 
 
 def test_build_ops():
```
Comment (math-fehr) [inline tests/dialects/test_pdl.py:14]: I have to say that this test is becoming a bit too large
Shouldn't we separate each constructor to a single test? That should help debugging and make things cleaner.

File: xdsl/dialects/pdl.py [hunk 3]
```diff
@@ -159,6 +172,32 @@ class OperationOp(Operation):
 
     irdl_options = [AttrSizedOperandSegments()]
 
+    @staticmethod
+    def get(opName: StringAttr | None,
+            attributeValueNames: ArrayAttr[StringAttr] | None = None,
+            operandValues: list[SSAValue] | None = None,
+            attributeValues: list[SSAValue] | None = None,
+            typeValues: list[SSAValue] | None = None):
```
Comment (math-fehr) [inline xdsl/dialects/pdl.py:180]: Nit: For this one and below, would `Iterable` be better?

**Review by math-fehr — APPROVED** [review]
Looks good to me!
Just added two nit comments

File: xdsl/dialects/builtin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (webmiche) [reply xdsl/dialects/builtin.py:244]: Actually, from this PR, it seems that there is another reason than just parsing and printing for this. Can you enlighten me, @superlopuh? Why was this change necessary for this PR?

**Pull request merged**